modules/cudastereo/include/opencv2/cudastereo.hpp

```diff
@@ -241,6 +241,41 @@ class CV_EXPORTS_W StereoConstantSpaceBP : public cuda::StereoBeliefPropagation
 CV_EXPORTS_W Ptr<cuda::StereoConstantSpaceBP>
     createStereoConstantSpaceBP(int ndisp = 128, int iters = 8, int levels = 4, int nr_plane = 4, int msg_type = CV_32F);
 
+/////////////////////////////////////////
+// StereoSGM
+
+/** @brief The class implements the modified H. Hirschmuller algorithm @cite HH08.
+Limitation and difference are as follows:
+
+-   Cost Aggregation use 8 paths only.
+-   Only quadratic interpolation is used for subpixel estimation in Disparity Computation.
+-   Disparity Refinement method can select only median filter with \f$3 \times 3\f$ window size.
+-   Mutual Information cost function is not implemented.
+Instead, Center-Symmetric Census Transform with \f$9 \times 7\f$ window size from @cite Spangenberg2013
+is used for robustness.
+
+@sa StereoSGBM
+*/
+class CV_EXPORTS_W StereoSGM : public cv::StereoSGBM
+{
+public:
+    CV_WRAP virtual void compute(InputArray left, InputArray right, OutputArray disparity) CV_OVERRIDE = 0;
+    CV_WRAP_AS(compute_s) virtual void compute(InputArray left, InputArray right, OutputArray disparity, Stream& stream) = 0;
+    CV_WRAP_AS(compute2) virtual void compute(InputArray left, InputArray right, OutputArray left_disp, OutputArray right_disp, Stream& stream) = 0;
```

Comment:
What should the name of the overload function be?

Reply:
In addition, I create new interface `compute2` bacause such a interface is implemented in OpenCV as I remember.
However, there is no interface in OpenCV and it make no sense.

Reply:
I'll fix it.